modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt

```diff
@@ -92,8 +89,8 @@ open class GenerateTask : DefaultTask() {
     /**
      * The Open API 2.0/3.x specification location.
      */
-    @get:InputFile
-    @PathSensitive(PathSensitivity.RELATIVE)
+    @Optional
+    @Input
```

Comment:
This would unfortunately mean that even if the input specification in a local file changed, the task would be up-to-date because it's only checking the filename but not its contents.

Maybe splitting the input specification location into local specs and remote specs would make sense?

Reply:
@joschi, if I extend this PR with the solution to check the modification status of the remote source will it do the trick? I mean let's imagine that the Up-to-date check works with the Url/Path then will you agree to use specificationUrl?